libvast/src/system/global_segment_store.cpp

```diff
@@ -0,0 +1,58 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/plugin.hpp"
+#include "vast/system/archive.hpp"
+#include "vast/system/node.hpp"
+#include "vast/system/node_control.hpp"
+
+namespace vast::system {
+
+// This store plugin wraps the global "archive" so we can
+// use a unified API in the transition period.
+class global_store_plugin final : public virtual store_plugin {
+public:
+  using store_plugin::builder_and_header;
+
+  // plugin API
+  caf::error initialize(data) override {
+    return {};
+  }
+
+  [[nodiscard]] const char* name() const override {
+    return "global_segment_store";
```

Comment:
TODO: Create slack  poll to decide on dashes vs. underscores as naming convention.